internal/repository/sync.go

```diff
@@ -62,8 +67,37 @@ func Clone(ctx context.Context, remoteURL, dir, initialBranch string) (*Reposito
 		return nil, errors.Join(ErrCloningRepository, err)
 	}
 
+	sort.Slice(expectedRootKeys, func(i, j int) bool {
```

Comment:
for readability, please move this into the if len > 0 block below.

Reply:
Fixed